lib/TPP/TileConsumerAndFuseProducers.cpp

```diff
@@ -33,6 +33,38 @@ using namespace mlir;
 
 namespace {
 
+struct ReplaceIterArgs : public OpRewritePattern<scf::ForOp> {
+  using OpRewritePattern<scf::ForOp>::OpRewritePattern;
+
+  void replaceIterArgs(PatternRewriter &rewriter, scf::ForOp outerFor,
+                       scf::ForOp innerFor) const {
+    assert(outerFor.getNumIterOperands() == innerFor.getNumIterOperands() &&
+           "expect same number of iter args");
+    Block *block = &(*innerFor.getRegion().begin());
+    for (auto it :
+         llvm::zip(outerFor.getIterOperands(), innerFor.getRegionIterArgs())) {
+      Value source = std::get<0>(it);
+      Value target = std::get<1>(it);
+      rewriter.replaceUsesWithIf(source, target, [&](OpOperand &use) {
```

Comment:
this looks awfully generic... without negative tests you can't be sure this won't hit patterns that it shouldn't

Reply:
Why is generic here?

Reply:
It seems this will replace the args of all loops in the contained function, even if they're not related to the pass you just ran, even if you didn't tile.

Reply:
No the pattern works only on tiled loop. See attributes checks above.  If other loops have the same issue we should apply this pattern anyway.